src/Database/Query/CacheableQueryBuilder.php

```diff
@@ -284,12 +290,25 @@ protected function log(string $message, string $level = 'debug')
             return false;
         }
 
-        if ($this->logLevel) {
-            Log::log($this->logLevel, "[Cacheable] {$message}");
-        } else {
-            Log::log($level, "[Cacheable] {$message}");
+        // Handle dynamic log level and log channel, so we can change it on the fly
+        // for example, if log channel is set to 'some_channel', it will log to 'some_channel' channel
+        // if log channel is set to 'default', it will log to the default channel
+        if ($this->logChannel == null || $this->logChannel == 'default') {
```

Comment:
Hi, these checks should be done once, and not with every `log` call. Please move such validations to the constructor and use default values in general. If you use the default value in the constructor and have `default` as default value anyway, these checks are too much. We should not work around wrong configs but encourage correctness.

Reply:
hi, thanks for feedback on my PR, just want to check, if i make logChannel default value as ```Log::getDefaultDriver``` and i move logChannel validation including array check on construct its that okey? so basically after prefilling some variable, i adding validation to check logChannel value, E.g

~~~bash
$this->logChannel = Arr::get($cacheableProperties, 'logging.channel', Log::getDefaultDriver());
$this->logEnabled = Arr::get($cacheableProperties, 'logging.enabled', false);
$this->logLevel = Arr::get($cacheableProperties, 'logging.level', 'debug');

if ($this->logChannel == 'default') $this->logChannel = Log::getDefaultDriver();

if (! in_array($this->logChannel, array_keys(config('logging.channels')))) {
        Log::log('error', "[Cacheable] Log channel '{$this->logChannel}' does not exist, using default driver instead.");
        $this->logChannel = Log::getDefaultDriver();
}
~~~

Let me know if this is okey for you, thanks

Reply:
Hey @Sleepy4k , here please add brackets, even if it is just a one-liner:
`if ($this->logChannel == 'default') $this->logChannel = Log::getDefaultDriver();`

I have made some second thoughts about this: I don't think I like this double check with `default` and would rather revert back to either it is `null`, then we go for the default driver, or the user has configured something, then we check. Checking against null and default is unnecessary in my opinion, just the absence is fine or a configuration, and no special case.

Thanks